src/background/services/storage.ts

```diff
@@ -126,23 +127,14 @@ export class StorageService {
     return false
   }
 
-  // TODO: ensure correct transitions between states, while also considering
-  // race conditions.
-  async setState(
-    state: null | Record<Exclude<Storage['state'], null>, boolean>
-  ): Promise<boolean> {
+  async setState(state: NonNullable<Storage['state']>): Promise<boolean> {
```

Comment:
Why does the function need to return a value?

Reply:
Unused for now, but can use it check if state was actually set.